app/vmselect/promql/rollup_test.go

```diff
@@ -1719,6 +1719,32 @@ func TestRollupDeltaWithStaleness(t *testing.T) {
 		timestampsExpected := []int64{0, 10e3, 20e3, 30e3, 40e3}
 		testRowsEqual(t, gotValues, rc.Timestamps, valuesExpected, timestampsExpected)
 	})
+
+	t.Run("issue-8935", func(t *testing.T) {
```

Comment:
Are there any additional test cases we should cover based on our understanding of the problem?